rubocop-airbnb/spec/rubocop/cop/airbnb/module_method_in_wrong_file_spec.rb

```diff
@@ -25,157 +23,120 @@
-      '      42',
-      '    end',
-      '  end',
-      'end',
-    ].join("\n")
-
     File.open "#{models_dir}/bar.rb", "w" do |file|
-      inspect_source(source, file)
+      expect_offense(<<~RUBY, file)
+        module Hello
+          module Foo
+            def baz
+            ^^^^^^^ In order for Rails autoloading to be able to find and load this file when someone calls this method, move the method definition to a file that defines the module. This file just uses the module as a namespace for another class or module. Method baz should be defined in hello/foo.rb.
+              42
+            end
+          end
+        end
+      RUBY
     end
-
-    expect(cop.offenses.size).to eq(1)
-    expect(cop.offenses.map(&:line).sort).to eq([3])
-    expect(cop.offenses.first.message).to include("Method baz should be defined in hello/foo.rb.")
   end
 
   it 'accepts if module method is in file with matching name' do
-    source = [
-      'module Foo',
-      '  def baz',
-      '    42',
-      '  end',
-      'end',
-    ].join("\n")
-
     File.open "#{models_dir}/foo.rb", "w" do |file|
-      inspect_source(source, file)
+      expect_no_offenses(<<~RUBY, file)
+        module Foo
+          def baz
+            42
+          end
+        end
+      RUBY
     end
-
-    expect(cop.offenses).to be_empty
   end
 
   it 'rejects with "self." static methods and a non-matching name' do
-    source = [
-      'module Foo',
-      '  def self.baz',
-      '    42',
-      '  end',
-      'end',
-    ].join("\n")
-
     File.open "#{models_dir}/bar.rb", "w" do |file|
-      inspect_source(source, file)
+      expect_offense(<<~RUBY, file)
+        module Foo
+          def self.baz
+          ^^^^^^^^^^^^ In order for Rails autoloading to be able to find and load this file when someone calls this method, move the method definition to a file that defines the module. This file just uses the module as a namespace for another class or module. Method (self) should be defined in foo.rb.
+            42
+          end
+        end
+      RUBY
     end
-
-    expect(cop.offenses.size).to eq(1)
-    expect(cop.offenses.map(&:line).sort).to eq([2])
   end
 
   it 'accepts with "self." static methods and a matching name' do
-    source = [
-      'module Foo',
-      '  def self.baz',
-      '    42',
-      '  end',
-      'end',
-    ].join("\n")
-
     File.open "#{models_dir}/foo.rb", "w" do |file|
-      inspect_source(source, file)
+      expect_no_offenses(<<~RUBY, file)
+        module Foo
+          def self.baz
+            42
+          end
+        end
+      RUBY
     end
-
-    expect(cop.offenses).to be_empty
   end
 
-  it 'rejects with "<<" static methods and a non-matching name' do
-    source = [
-      'module Foo',
-      '  class << self',
-      '    def baz',
-      '      42',
-      '    end',
-      '  end',
-      'end',
-    ].join("\n")
-
+  xit 'rejects with "<<" static methods and a non-matching name' do
```

Comment:
This test was set as pending due to the fact that a bug is causing it to fail, suggesting baz to be defined in foo/foo.rb instead of being in foo.rb.

Reply:
This seems strange to me. Why is the metaclass of `Foo` expected to be defined in `foo/foo.rb`? Was this failing previously? I'm assuming this is due to the rubocop upgrade?

I'd like to dig into this one and figure out why this changed

Reply:
Oh sorry I should have left the link to the [issue](https://github.com/airbnb/ruby/issues/190) that I created discussing this. It's a false positive in master.